### PR TITLE
GraphQL backend and frontend integration.

### PR DESCRIPTION
**Pull request by rmeena840**
GraphQL backend and frontend integration.

### CODE REVIEW

File: backend/server.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const QueryRoot = new GraphQLObjectType({
+	name: "Query",
+	fields: () => ({
+		Daos: {
+			type: new GraphQLList(DaoType),
+			description: "Dao Records",
+			resolve: async () => {
+				const res = await executeQuery(QUERY_GET_ALL_DAOS);
+				if (res && res.rows && res.rows.length) {
+					res.rows.map((item: any) => {
```
Comment (Megha-Dev-19) [inline backend/server.ts]: We can create a type of item since we know the structure of data

File: src/api/constants.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const BACKEND_URL =
+	process.env.BACKEND_URL || "http://localhost:4000/api";
```
Comment (Megha-Dev-19) [inline src/api/constants.ts:2]: api is little ambiguous here

File: src/api/fetch.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	return fetch(BACKEND_URL, {
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json",
+			Accept: "application/json",
+		},
+		body: JSON.stringify({
+			query,
+		}),
+	})
```
Comment (Megha-Dev-19) [inline src/api/fetch.ts:13]: Can you please explain me what we are doing here?

**Review by Megha-Dev-19 — COMMENTED** [review]
Good work. Left some comments

File: src/api/fetch.ts (hunk 3 above)
Comment (rmeena840) [reply src/api/fetch.ts:13]: We are trying to call the backend API here with the body being passed as a query param. Similar to what we do in axios.

File: src/api/constants.ts (hunk 2 above)
Comment (rmeena840) [reply src/api/constants.ts:2]: Graphql has only one endpoint.

File: backend/psql/constants.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const POSTGRES_UNAME = process.env.POSTGRES_UNAME || "algorand";
```
Comment (vuvoth) [inline backend/psql/constants.ts]: Should we rename it to `config.ts' ?

File: backend/server.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { executeQuery } from "./psql/queryExecute";
+import { QUERY_GET_ALL_DAOS } from "./psql/query";
+const express = require("express");
```
Comment (vuvoth) [inline backend/server.ts:3]: Shoud we change code to use `import..from` ?

File: backend/server.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+app.use(
+	"/api",
+	graphqlHTTP({
+		schema: schema,
+		graphiql: true,
+	})
+);
+
+app.listen(4000);
```
Comment (vuvoth) [inline backend/server.ts]: I think the port should be import from env.
```suggestion
app.listen(PORT, () => {
    console.log(`Backend is listening at ${PORT}`);
});
```

File: backend/server.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+		app_id: { type: GraphQLInt, description: "app id" },
+		app_params: {
+			type: GraphQLString,
+			description: "app params like global state etc.",
+		},
+		asset_id: { type: GraphQLInt, description: "asset id" },
+	}),
+});
+
+const QueryRoot = new GraphQLObjectType({
```
Comment (vuvoth) [inline backend/server.ts]: Move GraphQL define and config to new folder( how about move it to `graphql`).

**Review by vuvoth — COMMENTED** [review]
Nice work! 
I left few suggestions. 
Please check it! Thanks!

File: backend/server.ts (hunk 1 above)
Comment (rmeena840) [reply backend/server.ts]: `DoaItemType` has been added.

File: backend/server.ts (hunk 5 above)
Comment (rmeena840) [reply backend/server.ts:3]: No, it needs require.

File: backend/psql/constants.ts (hunk 4 above)
Comment (rmeena840) [reply backend/psql/constants.ts]: Done

File: backend/server.ts (hunk 6 above)
Comment (rmeena840) [reply backend/server.ts]: Good suggestion!!

File: backend/server.ts (hunk 7 above)
Comment (rmeena840) [reply backend/server.ts]: Done

File: backend/server.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { BACKEND_PORT } from "./config";
+import { schema } from "./graphql/schema";
+const express = require("express");
+const cors = require("cors");
+const app = express();
+const { graphqlHTTP } = require("express-graphql");
+const {
+	GraphQLObjectType,
+	GraphQLSchema,
```
Comment (vuvoth) [inline backend/server.ts]: Remove unused value here.

**Review by vuvoth — APPROVED** [review]
LGTM

File: backend/server.ts (hunk 8 above)
Comment (rmeena840) [reply backend/server.ts]: Oops, thx done

File: src/api/constants.ts (hunk 2 above)
Comment (Megha-Dev-19) [reply src/api/constants.ts:2]: okay, thanks for informing

**Pull request merged**